iceoryx_examples/automotive_soa/include/owl/kom/event_subscriber.inl

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_INL
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_INL
+
+#include "owl/kom/event_subscriber.hpp"
+
+namespace owl
+{
+namespace kom
+{
+template <typename T>
+inline EventSubscriber<T>::EventSubscriber(const core::String& service,
+                                           const core::String& instance,
+                                           const core::String& event) noexcept
+    : m_subscriber({service, instance, event},
+                   {QUEUE_CAPACITY, HISTORY_REQUEST, iox::NodeName_t(), NOT_OFFERED_ON_CREATE})
+{
+}
+
+template <typename T>
+inline void EventSubscriber<T>::Subscribe(std::size_t) noexcept
+{
+    // maxSampleCount is ignored, because it is an argument to the c'tor of m_subscriber as part of SubscriberOptions.
+    // Utilizing late initalization by wrapping m_subscriber in an cxx::optional and calling the c'tor here would be an
+    // option
+    m_subscriber.subscribe();
```

Comment:
As soon as the `m_subscriber` is attached to the listener you are not allowed to call it from the outside. The reason is that the listener can access `m_subscriber` at anytime from within its background thread and then you access this from here. The subscriber is not threadsafe and then pink elephants will dance around when you potentially access it from two threads at the same time.

Reply:
@mossmaurice 

> In the end it's up the users to use correctly IMHO.

Strongly disagree. Create an API which is easy to use and hard to misuse!
This is something where even I shot myself in the knee and I wrote this stuff. This is a design flaw in the listener at the moment and it will be solved in a way that the listener acquires the ownership of the attached objects. Then such code would lead to undefined behavior or a segfault.

Reply:
@elfenpiff Okidoki, you persuaded me. Due to the fact of late initialization for setting the `queueCapacity` the c'tor is now called in `Subscribe` and the d'tor in `Unsubscribed`. Additionally any receive handler that is set up is disabled. IMHO this is easier than other solutions were `m_subscriber` would still exist after calling `Unsubscribe`. Not super nice. If you have any advice for a better solution feel free to comment.